service/s3.h

```diff
@@ -261,9 +259,21 @@ struct S3Api : public AwsApi {
         std::string contentEncoding;
         std::map<std::string, std::string> metadata;
         std::string acl;
+    };
+
+    /** A set of options that are used when performing uploads. */
+    struct UploadOptions {
+        UploadOptions()
+            : numParallelRequests(8)
```

Comment:
Actually, the prototype of "upload" is the only place where I can see that -1 was specified by default, it meant that the default parameter for ObjectMetadata::numRequests should be used. And before being moved to UploadOptions, ObjectMetadata::numRequests was already set to 8 by default. I actually simplified that part of the code by moving "numRequests" out of the ObjectMetadata struct.